src/gui/DatabaseWidget.h

```diff
@@ -60,6 +59,8 @@ class DatabaseWidget : public QStackedWidget
     Q_OBJECT
 
 public:
+    friend class DatabaseOpenDialog;
```

Comment:
I don't think this is necessary, it is pointing to a larger issue with the unlockDatabase(bool) slot that is very overloaded with functionality.

Reply:
I knew you wouldn't like it. :laughing: 
I thought about just making unlockDatabase() public, but liked that even less. First, it's a bit overloaded with functionality and the signature is inconsistent with the other public slots (we could split it up to fix that), but secondly and most importantly, I don't want it to be part of the API. It's not something a normal user should ever be using, because it's more of an implementation detail and circumvents the normal unlocking procedure.